examples/boids/boids.comp

```diff
@@ -0,0 +1,79 @@
+#version 450
+
+// this shader expects NUM_PARTICLES and PARTICLES_PER_INVOCATION
```

Comment:
It looks like you are misunderstanding the computational model exposed by WebGPU/Vulkan. The shader `main()` function would work for a compute thread, it rarely need to loop over work items itself.
The shader also needs to define how many threads are ran per group, like:
```
layout(local_size_x = 64) in;
```
The default is 1, and this is the value used here for the lack of the explicit qualifier. This is in no way representing any practical workflow, since the threads in a group can share local data, which is what most programs try to take advantage of.

If you don't care what the thread group size needs to be, picking 64 is a safe choice, since it matches the largest wavefront length seen on the hardware.

Reply:
@kvark Thanks, I was definitely misunderstanding the parameters. With the change you suggested I now see the performance gains.